### PR TITLE
Add support for def_delegators (plural)

### PR DESCRIPTION
**Pull request by richardxia**
This PR builds on top of the constructs added in https://github.com/sorbet/sorbet/pull/3636 for handling `#def_delegator` by modifying the rewriter to also support [`#def_delegators`](https://ruby-doc.org/stdlib-2.5.1/libdoc/forwardable/rdoc/Forwardable.html#method-i-def_delegators), which is a variadic version of `#def_delegator`. My solution was to emit a synthetic delegator method for each of the methods passed to `#def_delegators`.

I split up the `DefDelegator::run()` into a few separate functions so that I could reuse the pieces that should be the same between `#def_delegator` and `#def_delegators`, but without trying to be overly DRY, since there are a few differences between the two. The biggest difference is that `#def_delegators` does not support the optional third argument of `#def_delegator`, which specifies an alias name for the delegator method. Another difference it is legal to call `#def_delegators` without specifying _any_ method names (e.g. `def_delegators :@foo`), and it just results in a no-op.

Because of these differences, I just had separate functions for the "argument parsing" of the two different methods, and I only shared the code for generating method stubs.

I'm not that experienced with C++, so apologies in advance if my code is not idiomatic, but I definitely welcome feedback on how to write this better. I especially don't really have a good feel for how ownership, references, and copies should should be done here.

Also, I apologize for not exactly following the instructions in CONTRIBUTING.md. If you want me to file a GitHub issue to discuss this more first, I can do that retroactively.

### Motivation

Sorbet already had an existing AST rewriter for `#def_delegator`, but not [`#def_delegators`](https://ruby-doc.org/stdlib-2.5.1/libdoc/forwardable/rdoc/Forwardable.html#method-i-def_delegators) (plural), which is a similar method in Ruby's `Forwardable` module.  `#def_delegators` is the same as `#def_delegator` except it takes a variable number of positional parameters, each specifying a different method symbol to delegate to another object. Since Sorbet already has basic support for `#def_delegator`, it would be convenient for it to support `#def_delegators` in a similar way.

### Test plan

I updated the tests that were first added in https://github.com/sorbet/sorbet/pull/3636 to include new test cases for `def_delegators`. I more or less applied the same test scenarios to `#def_delegators`.

### CODE REVIEW

**Comment by jez** [comment]
@jeffcarbs you added DefDelegators.cc. Can you look at this too?

**Comment by jeffcarbs** [comment]
I think that PR was my first time writing C++ 🙂 so I can't speak much to the code/refactoring. Based on the fact that none of the original tests were touched and they're all still passing I think this looks good to me 👍

**Comment by froydnj** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_JTaxR3koYs0zia
→ https://go/builds/bui_JTaxVo4HEVQdp2

File: rewriter/DefDelegator.cc [hunk 1]
```diff
@@ -47,26 +63,61 @@ vector<ast::ExpressionPtr> DefDelegator::run(core::MutableContext ctx, const ast
...
-        ast::MK::SyntheticMethod(loc, loc, methodName, std::move(args), ast::MK::RaiseUnimplemented(loc)));
+/// Handle #def_delegators for zero or more delegate methods
+vector<ast::ExpressionPtr> runDefDelegators(core::MutableContext ctx, const ast::Send *send) {
+    vector<ast::ExpressionPtr> methodStubs;
+    auto loc = send->loc;
 
-    // Include the original call to def_delegator so sorbet will still type-check it
+    // This method takes 1.. positional arguments and no keyword args:
+    // `def_delegators(accessor, methods*)`
+    if (!(send->numPosArgs >= 1)) {
```
Comment (froydnj) [inline rewriter/DefDelegator.cc:82]: Nit: I think this is more idiomatic as `send->numPosArgs == 0`.

**Review by froydnj — APPROVED** [review]
This looks good!  Can you push a fix for the tiny nit below?  I'll merge once that's done.

File: rewriter/DefDelegator.cc (hunk 1 above)
Comment (richardxia) [reply rewriter/DefDelegator.cc]: Oh yes, just amended my commit to make that change and force-pushed it back up.

**Comment by richardxia** [comment]
Thanks for reviewing!

**Pull request merged**